medusa/indexers/tmdb/tmdb.py

```diff
@@ -182,11 +182,27 @@ def search(self, series):
         """
         log.debug('Searching for show: {0}', series)
 
-        results = self._show_search(series, request_language=self.config['language'])
-        if not results:
+        results = None
+        # If search term is digit's only, store it and attempt to search by id.
+        show_by_id = None
+
+        try:
+            if series.isdigit():
+                show_by_id = self._get_show_by_id(series, request_language=self.config['language'])
+            results = self._show_search(series, request_language=self.config['language'])
+        except IndexerShowNotFound:
+            pass
+
+        if not results and not show_by_id:
             return
 
-        mapped_results = self._map_results(results, self.series_map, '|')
+        mapped_results = []
+        if results:
+            mapped_results = self._map_results(results, self.series_map, '|')
```

Comment:
Wouldn't it make more sense to just map the results in `_show_search` (aka `_get_show_by_name`)?